misc/random.c

```diff
@@ -83,3 +85,16 @@ double mp_rand_next_double(struct mp_rand_state *s)
 {
     return (mp_rand_next(s) >> 11) * 0x1.0p-53;
 }
+
+// Ref: https://www.pcg-random.org/posts/bounded-rands.html#bitmask-with-rejection-unbiased-apples-method
```

Comment:
Links in documentation IMO should include fallback snapshots to the wayback machine. It's very annoying to find a link in code documenting an algorithm, only to discover it's a deadlink.  Anyway, here: `https://web.archive.org/web/20250321082025/https://www.pcg-random.org/posts/bounded-rands.html#bitmask-with-rejection-unbiased-apples-method`